tools/project_manager/add_files.php

```diff
@@ -19,12 +19,22 @@
             font-family: ' . get_proofreading_font_family_fallback() . ';
         }
     ',
+    'js_data' => '
+        $(document).ready(function () {
+            $("#execute").submit(function (e) {
+                // disable the submit button
+                $("#submit").attr("disabled", true);
+                return true;
```

Comment:
This may make sense for a ton of our submits. Should we abstract it some?

Reply:
Abstract it such that all forms have their submit buttons disabled upon click or abstract it such that they _could_ be disabled without adding this to individual pages?

My gut says to not proactively abstract until we know of where else we want to use it so we know how best to abstract it. I'm open to ideas!

Reply:
Definitely the latter. Just so it is easy to enable for a given form if we think it is necessary, perhaps with some data- attribute. But we can cross that bridge perhaps when we encounter another place we think warrants this change. Especially if it needs to be reenabled under some circumstances such as client side errors.

Reply:
It would be useful for Remote File Manager's `Upload` button to appear disabled once it's pressed. There can be a significant delay between pressing the button and getting the message showing the progress of validity checking and virus checking.

Reply:
I thought that was fixed? In Chrome after I it Upload and the file is uploaded it immediately gives me a Upload status page as it's doing the checking. Maybe it works differently in different browsers. Regardless, we wouldn't be able to use this as-is for the RFM as the Upload button isn't a form Submit button.

Reply:
I'll check again tomorrow -- I thought I noticed a delay, but I may be wrong about that. That also may be the reason we didn't disable it at the time, come to think of it.

Reply:
I had a chance to look just now -- and since pressing the ```Upload``` button triggers a "Progress bar" showing the percentage, it shouldn't be an issue, so nevermind.

Reply:
I'm going to defer abstracting this until we have an obvious second need for it. I like the idea but I'm wary of adding the complexity without a second use-case.